wp-content/themes/workday/functions.php

```diff
@@ -97,4 +97,63 @@ function register_workday_widgets() {
 	}
 
 }
-add_action( 'widgets_init', 'register_workday_widgets', 1 );
\ No newline at end of file
+add_action( 'widgets_init', 'register_workday_widgets', 1 );
+
+/**
+ * Function that allows you to get all of the categories for a specific post
+ * with the top term returned as the first item in the array
+ * 
+ * @param int $post_id The ID of the post to get categories of
+ * @return Array of the categories for the specified post
+ */
+function workday_get_post_categories_with_top_term( $post_id ) {
+
+    // grab top term of post but don't link or echo it
+    $top_term = largo_top_term( $args = 
+        array(
+            'post' => $post_id,
+            'echo' => FALSE,
+            'link' => FALSE,
+        )
+    ); 
+
+    $formatted_categories = array();
+
+    // get all categories of the post
+    $category_list = get_the_category( $post_id );
+
+    // if our top term exists, let's move it to the front of our array
```

Comment:
If the top term is set and is a category, this moves the top term to the start of the list of categories.

If the top term is not set, this doesn't affect the category list.

How should this work if the top term is not a category?

Reply:
@benlk I think the top term has to be a category for this site... they're not using tags. Anything I'm missing?

Reply:
They're not using them, but they might use them in the future. They may decide to create tags or series, and use those as top terms, and this would not support those. Are we worried about that case?

Reply:
Not worried about tags, but yes to series...

Reply:
Let's do:
If the top term is set and is a category **or series,** this moves the top term to the start of the list of categories.

Reply:
We don't have a way of getting the top term term object from Largo (see https://github.com/INN/largo/issues/1775), so detecting whether the top term is a series would require parsing the `largo_top_term()` HTML response to determine whether it's a series.

It'd be easier to write something like:

> If the top term is set and is a category, this moves the top term to the start of the list of categories.
> If the top term is set, but the category list was not reordered, prepend the category list with the top term.
> If the top term is not set, do nothing special.
>
> Then, output the category list (whether or not it was modified)

Reply:
@benlk honestly I don't care about the order; just as long as all the categories are listed. if top term is first, that's great. and if the top term is a series, it should show up *somewhere* in the list.

Reply:
Resolved in https://github.com/INN/umbrella-workdayminn/pull/70#issuecomment-521052087